src/Components/ContactTypeIcon/ContactTypeIcon.jsx

```diff
@@ -1,42 +1,54 @@
 // @flow
 import * as React from "react";
 import TelegramIcon from "@skbkontur/react-icons/Telegram2";
-import DeviceSmartphoneIcon from "@skbkontur/react-icons/DeviceSmartphone";
 import MailIcon from "@skbkontur/react-icons/Mail2";
-import { ContactTypes } from "../../Domain/ContactType";
+import FacebookIcon from "@skbkontur/react-icons/Facebook2";
+import TwitterIcon from "@skbkontur/react-icons/Twitter2";
+import PhoneIcon from "@skbkontur/react-icons/Phone2";
 import SvgIcon from "../SvgIcon/SvgIcon";
 import PushoverLogo from "./pushover-logo.svg";
 import SlackLogo from "./slack-logo.svg";
 import TwilioLogo from "./twilio-logo.svg";
+import WebhookLogo from "./webhook-logo.svg";
+import WhatsappLogo from "./whatsapp-logo.svg";
 
 type Props = {
     type: string,
 };
 
 export default function ContactTypeIcon({ type }: Props): React.Element<any> {
-    if (type === ContactTypes.telegram) {
+    if (type.includes("slack")) {
```

Comment:
How do you think, can we refactored if-chain to some function?

Reply:
It's not clear to me what you mean by refactoring if-chain to some function but If we are talking about the alternatives to if-chain then switch case will be the most obvious alternative. Also we can transform it to a function and pass the string through a complex regex and get the desired outcome. But the latter will be more complex to understand for others

Reply:
I'm about something like this:
```js
const Validation = {
  telegram: (type) => Component | null,
  ...
};

const ContactIcon = (type) => {
  const Icon = Object.values(Validation).some(cb => cb(type));
  return <Icon />;
}
```

Reply:
Do you want to separate the validation logic from the React Element component?
Even if we're doing it this way, why can't we write a switch case or If-chain inside the function and return the Icon? Why should we write a separate function for each type and then map through every function to validate the icon type? Is it a convention or is this considered a good practice?

Reply:
I want to follow DRY principle. Look, all conditions like `type.include(some)`.

If we create some enum, like

```js
const TypeToIcon = {
  key: Component,
  ...
}
```

Then we can use it in cycle

```js
const ContactIcon = (type) => {
  const key = Object.keys(TypeToIcon).find((key) => type.includes(key));
  return key ? <TypeToIcon[key] /> : <Mail />; // or something like this
}
```

Reply:
Okay I get it. I'm working on it.

Reply:
I've made the changes. please review it @sashasushko